src/Jackalope/NodeType/NodeProcessor.php

```diff
@@ -161,12 +174,15 @@ private function processNodeWithType(NodeInterface $node, NodeType $nodeTypeDefi
             }
 
             if (!$node->hasProperty($propertyDef->getName())) {
-                if ($propertyDef->isMandatory() && !$propertyDef->isAutoCreated()) {
+                if ($propertyDef->isMandatory()
+                    && !$propertyDef->isAutoCreated()
+                    && $nodeTypeDefinition->getName() !== VersionHandler::MIX_VERSIONABLE
```

Comment:
see also https://github.com/jackalope/jackalope/pull/286#discussion_r30672101 - if this really is needed, i agree that we should move this check into a private method for better readability.